### PR TITLE
working pablo and accumulation for speech classification + correct arg parser so that flags overwrite the config used

### PR DESCRIPTION
**Pull request by csjiet**
Currently identifying a bug. Please do not merge.
**What does this accomplish?**: 
1) Allows USB's **Speech classification** pipeline to work with None, PabLO and accumulation methods, and fixes overwriting issue of config files. By fixing: **Data preprocessing and loading** in `semilearn/datasets/audio_datasets/datasetbase.py`, `semilearn/datasets/audio_datasets/pkl_data.py`, `semilearn/datasets/collactors/audio_collactor.py`
2) **Pre-logits extraction** from wave2vec speech classification model in`semilearn/nets/wave2vecv2/wave2vecv2.py`
3) **Adding extra `args` into function arguments for our calibrator model** to invoke pre-existing collators implemented in USB.
4) Adding code that allows the arg parser to take precedence over the config file
5) Add coverage calculation for fixmatch and flexmatch
6) Add CLIP embedding + MLP model for cifar10 (initial working implementation. Futurework: Optimize inference speed using better batching of training and unlabeled data during inference).

## PR review
Anyone in the community is welcome to review the PR.
Before you start reviewing, make sure you have read the [review guidelines](https://github.com/Lightning-AI/lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

### CODE REVIEW

File: confidence_funcs/calibration/auto_labeling_optimization_v0.py [hunk 1]
```diff
@@ -56,7 +56,7 @@ def __init__(self,clf, calib_conf,logger):
         self.logger = logger 
         self.g_model = None 
 
-    def init(self,ds_calib, clf_inf_out=None, use_prev_model=False ):
+    def init(self,args,ds_calib, clf_inf_out=None, use_prev_model=False ):
```
Comment (csjiet) [inline confidence_funcs/calibration/auto_labeling_optimization_v0.py:59]: `args` from the arg parser is passed from `algorithmbase.py` interface to relevant functions to allow for the collator in `semilearn/datasets/audio_datasets/pkl_data.py` to work with calibrator model.

File: confidence_funcs/classifiers/torch/clf_inference.py [hunk 2]
```diff
@@ -2,29 +2,38 @@
...
             
-        inference_conf.setdefault('batch_size',1024)
+        # TODO: Jiet: Set batch size of calibrator model
+        inference_conf.setdefault('batch_size', 1024)
         inference_conf.setdefault('shuffle',False)
         
         #inference_conf.setdefault('device','cpu')
         #self.logger.info("Running inference on {}".format(inference_conf['device']))
 
+        cf = None # collate_fn is None for vision datasets
```
Comment (csjiet) [inline confidence_funcs/classifiers/torch/clf_inference.py:26]: `cf`, short for the _collate function_ is passed as a keyword argument (`collate_fn=`) to Dataloader for the calibrator model

File: semilearn/datasets/audio_datasets/pkl_data.py [hunk 3]
```diff
@@ -64,6 +67,13 @@ def get_pkl_dset(args, alg='fixmatch', dataset='esc50', num_labels=40, num_class
         test_wav_list.append(bytes_to_array(test_data[idx]['wav']))
         test_label_list.append(int(test_data[idx]['label']))
```
Comment (csjiet) [inline semilearn/datasets/audio_datasets/pkl_data.py:69]: Data preprocessing for varying length raw audio waveforms at the dataset loading level. Can be removed in the future with proper use of collators.

File: semilearn/nets/wave2vecv2/wave2vecv2.py [hunk 4]
```diff
@@ -28,16 +31,32 @@ def forward(self, x, only_fc=False, only_feat=False, **kwargs):
...
             return logits
 
         pooled_output = self.extract(x)
 
         if only_feat:
             return pooled_output
 
-        logits = self.classifier(pooled_output)
-        result_dict = {'logits':logits, 'feat':pooled_output}
+        # logits = self.classifier(pooled_output)
```
Comment (csjiet) [inline semilearn/nets/wave2vecv2/wave2vecv2.py:47]: Extracting pre-logits from the `wave2vec` model

File: train.py [hunk 5]
```diff
@@ -222,7 +223,8 @@ def get_config():
     """
     Post-hoc calibration
     """
-    parser.add_argument("--use_post_hoc_calib", type=str2bool, default=False)
+    parser.add_argument("--use_post_hoc_calib", type=str2bool, default=False) 
+    parser.add_argument("--post_hoc_calib_conf_path", type=str)
```
Comment (csjiet) [inline train.py:227]: Allows explicit specification of post hoc calibration config created in `config/post-hoc` using the file identifier.

File: train.py [hunk 6]
```diff
@@ -276,7 +277,8 @@ def get_config():
     # add algorithm specific parameters
     args = parser.parse_args()
     
-    over_write_args_from_file(args, args.c)
+    # over_write_args_from_file(args, args.c)
+    over_write_file_from_args(args, args.c) # args take precendence
```
Comment (csjiet) [inline train.py:281]: Function that only takes declared args from config file if it is not already in the arg parser
